apiary-ranger-metastore-plugin/src/test/java/com/expedia/apiary/extensions/rangerauth/listener/RangerAdminClientImpl.java

```diff
@@ -0,0 +1,88 @@
+/**
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.expedia.apiary.extensions.rangerauth.listener;
+
+import java.io.File;
+import java.nio.file.FileSystems;
+import java.nio.file.Files;
+import java.util.List;
+
+import org.apache.ranger.admin.client.RangerAdminClient;
+import org.apache.ranger.plugin.util.GrantRevokeRequest;
+import org.apache.ranger.plugin.util.ServicePolicies;
+import org.apache.ranger.plugin.util.ServiceTags;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+
+/**
+ * A test implementation of the RangerAdminClient interface that just reads policies in from a file and returns them
+ */
+public class RangerAdminClientImpl implements RangerAdminClient {
```

Comment:
Would a better name be `PolicyFileRangerAdminClient`? `Impl` doesn't really say anything.

Reply:
This is a copy from ranger project hive-agent unit tests
https://github.com/apache/ranger/blob/master/hive-agent/src/test/java/org/apache/ranger/services/hive/RangerAdminClientImpl.java

not sure how we handle in these cases, do you still wan't me to make changes recomended or keep it as close as possible to the source?

Reply:
Ah, OK, I see now. In this case you need to update the copyright header to clearly state where the other code here comes from. I'll add this to this PR quickly as it also usually requires a change to the license generation plugin so we don't overwrite the copyright on files which have code from multiple sources. If this came from somewhere else I see the rationale for keeping it as similar to the upstream source as possible so we can leave it named like this.